File: apps/desktop/src/api/workspace.ts
import { invoke } from '@tauri-apps/api/core';
import type {
  CreateWorkspaceRequest,
  CreateWorkspaceResult,
  WorkspaceTemplateSummary,
} from '@features/workspace-wizard/types';

// Workspace Tauri command stubs — real implementations land with lore-workspaces crate

export async function openWorkspace(path: string): Promise<void> {
  return invoke('open_workspace', { path });
}

export async function listWorkspaceTemplates(): Promise<WorkspaceTemplateSummary[]> {
  return invoke('list_workspace_templates');
}

export async function createWorkspace(
  request: CreateWorkspaceRequest,
): Promise<CreateWorkspaceResult> {
  return invoke('create_workspace', { request });
}

export async function listRecentWorkspaces(): Promise<unknown[]> {
  return invoke('list_recent_workspaces');
}

File: apps/desktop/src-tauri/src/commands/workspace.rs
use lore_workspaces::{
    CreateWorkspaceRequest, CreateWorkspaceResult, WorkspaceTemplateSummary,
    create_workspace as create_workspace_impl, list_workspace_templates as list_templates_impl,
};

#[tauri::command]
pub fn list_workspace_templates() -> Result<Vec<WorkspaceTemplateSummary>, String> {
    Ok(list_templates_impl())
}

#[tauri::command]
pub fn create_workspace(request: CreateWorkspaceRequest) -> Result<CreateWorkspaceResult, String> {
    create_workspace_impl(request).map_err(|error| error.to_string())
}
